Code/GraphMol/FindRings.cpp

```diff
@@ -745,15 +747,17 @@ bool _atomSearchBFS(const ROMol &tMol, unsigned int startAtomIdx,
     while (nbrIdx != endNbrs) {
       if (*nbrIdx == endAtomIdx) {
         if (currAtomIdx != startAtomIdx) {
-          tv.push_back(rdcast<unsigned int>(*nbrIdx));
+          INT_VECT nv(tv);
```

Comment:
`tv` was potentially being modified on every pass through the loop starting at line 747. This is wrong and was leading to problems in rare cases. The hyper-connected PDB file that lead to the bug report happened to be one of those.